cmake/compilers/AppleClang.cmake

```diff
@@ -37,7 +37,7 @@ if (CMAKE_OSX_ARCHITECTURES)
 else()
     set(_tbb_target_architectures "${CMAKE_SYSTEM_PROCESSOR}")
 endif()
-if ("${_tbb_target_architectures}" MATCHES "(x86_64|amd64|AMD64)") # OSX systems are 64-bit only
+if ("${_tbb_target_architectures}" MATCHES "(i.86|x86|x86_64|amd64|AMD64)") # Clang is not supported on PowerPC
```

Comment:
I don't quite understand how this line "Clang is not supported on PowerPC" is related to this arch check? Does it mean that when configuring project on PPC `COMPILER_ID` would be `AppleClang` and not `Clang` so 32 bit check must be included here?

Reply:
I agree. If I understand correctly, this code isn't run on systems without Clang support anyway, is it?

Reply:
just to clarify (I do not think it makes any material difference to the outcome of the PR).

Xcode clang on 10.6 *does* accept `-arch ppc` but the clang driver just invokes `/usr/bin/gcc` to handle it (i.e. there is no PPC clang implementation there)/

```
$ clang /source/test/ABI/pr110044.c -arch ppc -v -S
Apple clang version 1.7 (tags/Apple/clang-77) (based on LLVM 2.9svn)
Target: powerpc-apple-darwin10
Thread model: posix
clang: warning: not using the clang compiler for the 'powerpc' architecture
 "/usr/bin/gcc" -v -S -arch ppc -m32 -o pr110044.s -x c /source/test/ABI/pr110044.c
Using built-in specs.
```